### PR TITLE
Tests for News

### PR DESCRIPTION
**Pull request by VolodymyrTrach**

### CODE REVIEW

File: src/store/store.js [hunk 1]
```diff
@@ -12,9 +12,9 @@ const configureStore = () => {
   const store = createStore(
     rootReducer(history),
     compose(
-      applyMiddleware(routerMiddleware(history), sagaMiddleware),
-      window.__REDUX_DEVTOOLS_EXTENSION__ &&
-        window.__REDUX_DEVTOOLS_EXTENSION__()
+      applyMiddleware(routerMiddleware(history), sagaMiddleware)
+      // window.__REDUX_DEVTOOLS_EXTENSION__ &&
+      //   window.__REDUX_DEVTOOLS_EXTENSION__()
```
Comment (bandvov) [inline src/store/store.js:17]: you should uncoment redux devtools

File: src/redux/news/tests/news.saga.test.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+            },
+            date: '154568794567',
+            title: 'Test title',
+            text: 'Test text'
+          }
+        }
+      }
+    };
+
+    return expectSaga(handleArticleLoad, { payload: '13546789456' })
```
Comment (ecl1pseo) [inline src/redux/news/tests/news.saga.test.js:65]: I guess you should also write test in the case if we have some errors from server

Comment (VolodymyrTrach) [reply src/redux/news/tests/news.saga.test.js:83]: added

File: src/store/store.js (hunk 1 above)
Comment (VolodymyrTrach) [reply src/store/store.js]: done

File: src/redux/news/news.reducer.js [hunk 3]
```diff
@@ -6,7 +6,7 @@ const initialState = {
   activeArticle: null
 };
 
-const newsReducer = (state = initialState, action) => {
+const newsReducer = (state = initialState, action = {}) => {
```
Comment (Alan18081) [inline src/redux/news/news.reducer.js:9]: It is useless default parameter

File: src/redux/news/news.sagas.js [hunk 4]
```diff
@@ -89,7 +89,7 @@ function* handleArticleLoad({ payload }) {
   }
 }
 
-function* handleNewsError(e) {
+export function* handleNewsError(e) {
```
Comment (Alan18081) [inline src/redux/news/news.sagas.js:92]: Why have you added  `export` here? This saga function should be used only inside this file and root news saga

File: src/redux/news/news.sagas.js [hunk 5]
```diff
@@ -47,7 +47,7 @@ function* handleNewsLoad() {
   }
 }
 
-function* handleArticleLoad({ payload }) {
+export function* handleArticleLoad({ payload }) {
```
Comment (Alan18081) [inline src/redux/news/news.sagas.js:50]: Why have you added  `export` here? This saga function should be used only inside this file and root news saga

File: src/redux/news/news.sagas.js [hunk 6]
```diff
@@ -5,7 +5,7 @@ import { setError } from '../error/error.actions';
 import getItems from '../../utils/client';
 import { GET_NEWS, GET_NEWS_ARTICLE } from './news.types';
 
-function* handleNewsLoad() {
+export function* handleNewsLoad() {
```
Comment (Alan18081) [inline src/redux/news/news.sagas.js:8]: Why have you added  `export` here? This saga function should be used only inside this file and root news saga

File: src/redux/news/tests/news.saga.test.js [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+            }
+          ]
+        }
+      }
+    };
+
+    return expectSaga(handleNewsLoad)
+      .provide([[matchers.call.fn(getItems), newsExample]])
+      .put({ type: SET_LOADING, payload: true })
+      .put({
```
Comment (Alan18081) [inline src/redux/news/tests/news.saga.test.js:41]: Please, use action creator here

File: src/redux/news/tests/news.saga.test.js [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+        type: SET_NEWS,
+        payload: newsExample.data.getAllNews
+      })
+      .put({ type: SET_LOADING, payload: false })
```
Comment (Alan18081) [inline src/redux/news/tests/news.saga.test.js:45]: Please, use action creator here

File: src/redux/news/tests/news.saga.test.js [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+      .run();
+  });
+});
+
+it('handles errors', () => {
+  const e = new Error('news not found');
+
+  return expectSaga(handleNewsLoad)
+    .provide([[matchers.call.fn(getItems), throwError(e)]])
+    .put({ type: SET_LOADING, payload: true })
```
Comment (Alan18081) [inline src/redux/news/tests/news.saga.test.js:55]: Please, use action creator here

File: src/redux/news/tests/news.saga.test.js [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+    .put({ type: SET_LOADING, payload: false })
```
Comment (Alan18081) [inline src/redux/news/tests/news.saga.test.js:56]: Please, use action creator here

File: src/redux/news/tests/news.saga.test.js [hunk 11]
```diff
@@ -0,0 +1,100 @@
...
+    .put({ type: SET_ERROR, payload: { e } })
```
Comment (Alan18081) [inline src/redux/news/tests/news.saga.test.js:57]: Please, use action creator here

File: src/redux/news/tests/news.saga.test.js [hunk 12]
```diff
@@ -0,0 +1,100 @@
...
+        type: SET_NEWS,
+        payload: newsExample.data.getAllNews
+      })
+      .put({ type: SET_LOADING, payload: false })
+      .run();
+  });
+});
+
+it('handles errors', () => {
+  const e = new Error('news not found');
```
Comment (Alan18081) [inline src/redux/news/tests/news.saga.test.js:51]: Please, remove this redundant variable and inline creation in the action creator

File: src/redux/news/tests/news.saga.test.js [hunk 13]
```diff
@@ -0,0 +1,100 @@
...
+            title: 'Test title',
+            text: 'Test text'
+          }
+        }
+      }
+    };
+
+    return expectSaga(handleArticleLoad, { payload: '13546789456' })
+      .provide([[matchers.call.fn(getItems), articleExample]])
+      .put({ type: SET_LOADING, payload: true })
```
Comment (Alan18081) [inline src/redux/news/tests/news.saga.test.js:81]: Please, use action creator here

File: src/redux/news/tests/news.saga.test.js [hunk 14]
```diff
@@ -0,0 +1,100 @@
...
+      .put({
```
Comment (Alan18081) [inline src/redux/news/tests/news.saga.test.js:82]: Please, use action creator here

File: src/redux/news/tests/news.saga.test.js [hunk 15]
```diff
@@ -0,0 +1,100 @@
...
+        type: SET_NEWS_ARTICLE,
+        payload: articleExample.data.getNewsById
+      })
+      .put({ type: SET_LOADING, payload: false })
```
Comment (Alan18081) [inline src/redux/news/tests/news.saga.test.js:86]: Please, use action creator here

File: src/redux/news/tests/news.reducer.test.js [hunk 16]
```diff
@@ -0,0 +1,115 @@
...
+    const state = {
+      loading: true,
+      list: newsExample,
+      activeArticle: null
+    };
+
+    expect(newsReducer(initialState, setNews(newsExample))).toEqual(state);
+  });
+
+  it('should return news array', () => {
```
Comment (Alan18081) [inline src/redux/news/tests/news.reducer.test.js:65]: Why do you test action that is not handled by reducer in `switch -> case`

File: src/redux/news/tests/news.reducer.test.js [hunk 17]
```diff
@@ -0,0 +1,115 @@
...
+      list: [],
+      activeArticle: articleExample
+    };
+
+    expect(newsReducer(initialState, setArticle(articleExample))).toEqual(
+      state
+    );
+  });
+
+  it('should return aticle', () => {
```
Comment (Alan18081) [inline src/redux/news/tests/news.reducer.test.js:87]: Why do you test action that is not handled by reducer in `switch -> case`

File: src/redux/news/news.reducer.js (hunk 3 above)
Comment (VolodymyrTrach) [reply src/redux/news/news.reducer.js:9]: I know that it is useless. Sonarcloud says that it is code smell without it. If it is ok to leave this code smell, I`ll remove it

File: src/redux/news/tests/news.saga.test.js (hunk 7 above)
Comment (VolodymyrTrach) [reply src/redux/news/tests/news.saga.test.js:41]: done

File: src/redux/news/tests/news.saga.test.js (hunk 11 above)
Comment (VolodymyrTrach) [reply src/redux/news/tests/news.saga.test.js:57]: done

File: src/redux/news/tests/news.saga.test.js (hunk 13 above)
Comment (VolodymyrTrach) [reply src/redux/news/tests/news.saga.test.js:81]: done

File: src/redux/news/tests/news.saga.test.js (hunk 15 above)
Comment (VolodymyrTrach) [reply src/redux/news/tests/news.saga.test.js:86]: done

File: src/redux/news/tests/news.saga.test.js (hunk 14 above)
Comment (VolodymyrTrach) [reply src/redux/news/tests/news.saga.test.js:82]: done

File: src/redux/news/tests/news.saga.test.js (hunk 8 above)
Comment (VolodymyrTrach) [reply src/redux/news/tests/news.saga.test.js:45]: done

File: src/redux/news/tests/news.saga.test.js (hunk 9 above)
Comment (VolodymyrTrach) [reply src/redux/news/tests/news.saga.test.js:55]: done

File: src/redux/news/tests/news.saga.test.js (hunk 10 above)
Comment (VolodymyrTrach) [reply src/redux/news/tests/news.saga.test.js:56]: done

File: src/redux/news/tests/news.reducer.test.js (hunk 16 above)
Comment (VolodymyrTrach) [reply src/redux/news/tests/news.reducer.test.js:65]: removed

File: src/redux/news/tests/news.reducer.test.js (hunk 17 above)
Comment (VolodymyrTrach) [reply src/redux/news/tests/news.reducer.test.js:87]: removed

File: src/redux/news/news.sagas.js (hunk 4 above)
Comment (VolodymyrTrach) [reply src/redux/news/news.sagas.js:92]: I exported it for tests

File: src/redux/news/news.sagas.js (hunk 5 above)
Comment (VolodymyrTrach) [reply src/redux/news/news.sagas.js:50]: I exported it for tests

File: src/redux/news/news.sagas.js (hunk 6 above)
Comment (VolodymyrTrach) [reply src/redux/news/news.sagas.js:8]: I exported it for tests

File: src/redux/news/tests/news.saga.test.js (hunk 12 above)
Comment (VolodymyrTrach) [reply src/redux/news/tests/news.saga.test.js:49]: I simulate error here

**Pull request merged**